libs/pilight/protocols/433.92/smartwares_switch.c

```diff
@@ -0,0 +1,277 @@
+		return EXIT_FAILURE;
+	} else if((unit > 15 || unit < 0) && all == 0) {
+		logprintf(LOG_ERR, "smartwares_switch: invalid unit range");
+		return EXIT_FAILURE;
+	} else {
+		if(unit == -1 && all == 1) {
+			unit = 0;
+		}
+		createMessage(id, unit, state, all, learn);
+		createStart();
+		clearCode();
+		createId(id);
+		createAll(all);
+		createState(state);
+		createUnit(unit);
+		createFooter();
+		smartwares_switch->rawlen = RAW_LENGTH;
+	}
+	return EXIT_SUCCESS;
+}
+
+static void printHelp(void) {
+	printf("\t -t --on\t\t\tsend an on signal\n");
+	printf("\t -f --off\t\t\tsend an off signal\n");
+	printf("\t -u --unit=unit\t\t\tcontrol a device with this unit code\n");
+	printf("\t -i --id=id\t\t\tcontrol a device with this id\n");
+	printf("\t -a --all\t\t\tsend command to all devices with this id\n");
+	printf("\t -l --learn\t\t\tsend multiple streams so switch can learn\n");
+}
+
+#if !defined(MODULE) && !defined(_WIN32)
+__attribute__((weak))
+#endif
+void smartwaresSwitchInit(void) {
+	protocol_register(&smartwares_switch);
+	protocol_set_id(smartwares_switch, "smartwares_switch");
+	protocol_device_add(smartwares_switch, "smartwares_switch", "Smartwares Switches");
+	smartwares_switch->devtype = SWITCH;
+	smartwares_switch->hwtype = RF433;
+	smartwares_switch->txrpt = NORMAL_REPEATS;
+	smartwares_switch->minrawlen = RAW_LENGTH;
+	smartwares_switch->maxrawlen = RAW_LENGTH;
+	smartwares_switch->maxgaplen = MAX_PULSE_LENGTH*PULSE_DIV;
+	smartwares_switch->mingaplen = MIN_PULSE_LENGTH*PULSE_DIV;
+
+	options_add(&smartwares_switch->options, "t", "on", OPTION_NO_VALUE, DEVICES_STATE, JSON_STRING, NULL, NULL);
+	options_add(&smartwares_switch->options, "f", "off", OPTION_NO_VALUE, DEVICES_STATE, JSON_STRING, NULL, NULL);
+	options_add(&smartwares_switch->options, "u", "unit", OPTION_HAS_VALUE, DEVICES_ID, JSON_NUMBER, NULL, "^([0-9]{1}|[1][0-5])$");
+	options_add(&smartwares_switch->options, "i", "id", OPTION_HAS_VALUE, DEVICES_ID, JSON_NUMBER, NULL, "^([0-9]{1,7}|[1-5][0-9]{7}|6([0-6][0-9]{6}|7(0[0-9]{5}|10([0-7][0-9]{3}|8([0-7][0-9]{2}|8([0-5][0-9]|6[0-3]))))))$");
+	options_add(&smartwares_switch->options, "a", "all", OPTION_OPT_VALUE, DEVICES_OPTIONAL, JSON_NUMBER, NULL, NULL);
+	options_add(&smartwares_switch->options, "l", "learn", OPTION_NO_VALUE, DEVICES_OPTIONAL, JSON_NUMBER, NULL, NULL);
+
+	options_add(&smartwares_switch->options, "0", "readonly", OPTION_HAS_VALUE, GUI_SETTING, JSON_NUMBER, (void *)0, "^[10]{1}$");
+	options_add(&smartwares_switch->options, "0", "confirm", OPTION_HAS_VALUE, GUI_SETTING, JSON_NUMBER, (void *)0, "^[10]{1}$");
+
+	smartwares_switch->parseCode=&parseCode;
+	smartwares_switch->createCode=&createCode;
+	smartwares_switch->printHelp=&printHelp;
+	smartwares_switch->validate=&validate;
+}
+
+#if defined(MODULE) && !defined(_WIN32)
+void compatibility(struct module_t *module) {
+	module->name = "smartwares_switch";
+	module->version = "1.0";
+	module->reqversion = "1.0";
```

Comment:
Can you fix these values?

Reply:
Sure, I'll give it a try as I'm not 100% sure I understand those values. In case I got it wrong, please suggest what values I should use.